pallets/nodle-staking/src/migrations.rs

```diff
@@ -113,8 +153,22 @@ pub fn poa_validators_migration<T: Config>() -> Weight {
                 <StakingInvulnerables<T>>::put(post_invulnerable_validators.clone());
                 weight = weight.saturating_add(RocksDbWeight::get().reads_writes(0, 1));
             } else {
-                log::error!("Error!!! Staking Invulnerable invalid DB instance state");
-                return weight;
+                log::info!("Staking runtime upgrade Genesis config");
+
+                <StakingInvulnerables<T>>::put(poa_validators.clone());
+
+                // Ensure balance is >= ED
+                let imbalance = T::Currency::issue(T::Currency::minimum_balance());
+                T::Currency::resolve_creating(&T::PalletId::get().into_account(), imbalance);
```

Comment:
I would like to avoid creating coins if possible due to the fact we are supposed to have a fixed supply. I would suggest checking wether we really need to do that, if we really need to let's take the funds from the company reserve instead

Reply:
Indeed not required :+1: